pkg/backend/display/object_diff.go

```diff
@@ -1215,13 +1220,23 @@ func (p *propertyPrinter) decodeValue(repr string) (resource.PropertyValue, stri
 			return nil, "", false
 		}
 
-		r.Reset(repr)
-		if err := yaml.NewDecoder(r).Decode(&object); err == nil {
-			translated, ok := p.translateYAMLValue(object)
-			if !ok {
-				return nil, "", false
+		pos, err := r.Seek(0, io.SeekStart)
+		contract.Assertf(err == nil, "strings.NewReader.Seek(0, io.SeekStart) failed: %v", err)
+		contract.Assertf(pos == 0, "strings.NewReader.Seek(0, io.SeekStart) failed: pos=%d", pos)
+
+		yamlDecoder := yaml.NewDecoder(r)
+		if err := yamlDecoder.Decode(&object); err == nil {
+			// Make sure _all_ the string was consumed as YAML. Unlike JsonDecoder above, the YamlDecoder
+			// doesn't give an easy way to do this, so our workaround is we ask it to try and decode another
+			// value, and if it fails with io.EOF, then we know we've consumed the whole string.
+			eofErr := yamlDecoder.Decode(nil)
```

Comment:
Should this be a non-nil `*any` (`new(any))`) if we want the YAML decoder to try to decode anything that follows?

Reply:
Decode is happy just having nil here, it just skips setting the value but still returns if there's EOF or more data which is all we care about here.